std/fs/write_json.ts

```diff
@@ -7,6 +9,21 @@ export interface WriteJsonOptions {
   replacer?: Array<number | string> | Replacer;
 }
 
+function createJsonWithEol(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  object: any,
+  options: WriteJsonOptions,
+): string {
+  const contentRaw = JSON.stringify(
+    object,
+    options.replacer as string[],
+    options.spaces,
+  );
+
+  const eol = Deno.build.os === "windows" ? EOL.CRLF : EOL.LF;
+  return format(`${contentRaw}\n`, eol);
```

Comment:
This is a breaking change from current behavior. Currently, LF is used for all newlines, regardless of OS environment.

If newline replacement/formatting is being considered, it should be optional and not the default behavior.

Reply:
On the topic of options:

> so it should decide internally.

@nayeemrmn Is that what you meant? A new option in `WriteJsonOptions`?

Reply:
I'm not sure what the convention is for such options, something like this?
```typescript
enum WriteJsonEol {
  LF,
  PlatformSpecific
}

export interface WriteJsonOptions {
  ..,
  platform?: WriteJsonEol;
}
```

Reply:
Or maybe
```typescript
type WriteJsonEol = "LF" | "PlatformSpecific";
```

Reply:
@tranzystorek-io See https://github.com/tranzystorek-io/deno/pull/1

Reply:
> This is a breaking change from current behavior. Currently, LF is used for all newlines, regardless of OS environment.
>
> If newline replacement/formatting is being considered, it should be optional and not the default behavior.

It doesn't matter if it's breaking because `std` is unstable. I'm not sure what you mean here, currently there is no new line, right? It's not breaking either way because the spec of the function is to write a valid JSON string to a file. Not sure it needs to be configurable, is there any reason not to write the new line?

Reply:
> I'm not sure what you mean here, currently there is no new line, right? It's not breaking either way because the spec of the function is to write a valid JSON string to a file.

@nayeemrmn If a value is provided in the `spaces` option, it gets passed to `JSON.stringify` and the resulting string includes that whitespace, inserting LF newlines in the formatted JSON string.

Reply:
By "breaking" I mean breaking expectations, not breaking in the `semver` sense. See the example below; the result should be the same in all OS environments:

```js
(() => {
  const CR = '\r';
  const LF = '\n';

  const spaces = 2;
  const data = { a: 1, b: 2 };
  const json = JSON.stringify(data, null, spaces);

  console.log(json);

  console.log({
    CR: json.includes(CR), // false
    LF: json.includes(LF), // true
  });
})();
```